### ISSUE TITLE
Image load in Edge versions with partial support 

### ISSUE THREAD

**Issue by DKvistgaard** [constancecchen/object-fit-polyfill#51]
We're experiencing some issues using this (otherwise fantastic polyfill) in Edge browsers with partial support (versions 16, 17 and 18). The issue is that it will incorrectly calculate the object-fit placement of the images on page load in these browsers. But as soon as the user resizes the window, then it repositions perfectly.

So, I looked through the code to check how everything works and see if we were missing something or and an incorrect setup. In the code I found the setup where it waits for the image to load before doing the calculations, but this doesn't trigger for Edge browsers with partial support. This is due to the if-statement: 

```js
if (mediaType === 'img' && !edgePartialSupport) {
    if (media[i].complete) {
        objectFit(media[i]);
    } else {
        media[i].addEventListener('load', function() {
            objectFit(this);
        });
    }
}
```

Is there any reason why this is omitted if the Edge browser has partial support? It still runs all the same placement calculations as for Edge browsers with no support, so I can't see why this should include all browsers with partial support as well.

I've added a pull request to include Edge browsers with partial support as well, but I simply wanted to be 100% sure, that there isn't a good reason that I haven't thought of.

I also see this happening on the [demo site](https://constance.dev/object-fit-polyfill/demo/) in Edge (just remember to clear cache). See a screenshot of the error here:
![Screenshot 2019-06-25 at 16 04 15](https://user-images.githubusercontent.com/1705203/60105158-1a481980-9763-11e9-982f-d3c630cf2fa3.png)

**Comment by cee-chen** [comment]
Hey hey! I'm not quite sure we're on the same page with understanding what that piece of code is doing. It has nothing to do with whether an whether the image has loaded or not. **The polyfill simply shouldn't be running on images, period**, because Edge supports `object-fit` as-is for images but not for videos/other elements - at least, that was the case for 16 and 17 - I haven't checked yet for 18 and will do so here shortly. (Also see #33 and #35 for the original issues with Edge.)

So there's nothing for the polyfill to do, and it _should_ skip all images entirely. I'll check in here to see what else could be going be wrong.

**Comment by cee-chen** [comment]
Ah dangit. I see where I messed up. I'm so sorry, this is sloppy branching on my part when I added support for canvas/generic elements (#43 / cf24a3e3fb4a1d9e256478c0d64dcff0721f9e78). I'll have a fix out here shortly.

**Comment by cee-chen** [comment]
Okay - this should have been fixed by 746d10293e26a117917af72e6819ff6801ad8b32.

So now for Edge, images should be untouched/skipped entirely by the polyfill but video, canvas etc. should still retain the polyfill styling. See below example:

![d6600a4f-6b17-42e7-92c3-826117fc8e2f](https://user-images.githubusercontent.com/549407/60205148-4a47f900-9805-11e9-8eeb-acf221fc9872.png)

Sorry again for the issue and thanks for your patience. On the plus side [caniuse is showing that Edge will eventually contain full object-fit support](https://caniuse.com/#feat=object-fit)... so at some point we can put this shenanigans behind us. :)

**Comment by DKvistgaard** [comment]
@constancecchen Ah yes, much better. I also wondered why the `img` elements where even touched in the first place in Edge browsers, so I thought there was a good reason behind it. So this is a much better fix.

Thanks 😄
